### PR TITLE
AF-2404 : Package dropdown button is not visible (slowly appears) when creating an asset

### PR DESCRIPTION
**Pull request by akumar074**
* Fixed invisible package dropdown by setting it up on PostConstruct in KieSelectElement.

 * Moved unnecessary multiple callbacks to one for PackageListBox.

 * Removed parent container of KieSelectElement to be used directly.
 
 * Resetting package list dropdown to fix dropdown with previous application package details.

Ensemble:
jbpm-wb [#1409](https://github.com/kiegroup/jbpm-wb/pull/1409)

Deployable WAR Link: https://drive.google.com/open?id=11j9krqVd5xgBhoGw_sMBxAkGICkbzYTj

### CODE REVIEW

File: kie-wb-common-services/kie-wb-common-services-api/src/main/java/org/kie/workbench/common/services/shared/project/KieModulePackages.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package org.kie.workbench.common.services.shared.project;
```
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-services-api/src/main/java/org/kie/workbench/common/services/shared/project/KieModulePackages.java]: Please add a license.

**Comment by manstis** [comment]
@paulovmr IDK if you're the best to review; but someone from _core_ should.

I also don't know who from QE would be responsible for a review... can you please assign someone?

File: kie-wb-common-services/kie-wb-common-services-api/src/main/java/org/kie/workbench/common/services/shared/project/KieModulePackages.java (hunk 1 above)
Comment (akumar074) [reply kie-wb-common-services/kie-wb-common-services-api/src/main/java/org/kie/workbench/common/services/shared/project/KieModulePackages.java]: Thank you @manstis for reporting this. I have fixed this with other changes.

**Comment by akumar074** [comment]
jenkins execute full downstream build please

**Comment by akumar074** [comment]
jenkins execute full downstream build please

**Comment by akumar074** [comment]
Jenkins retest please

**Comment by akumar074** [comment]
jenkins execute full downstream build please

**Comment by akumar074** [comment]
jenkins execute full downstream build please

**Comment by ederign** [comment]
@caponetto @barboras7 could you please take a look at this?

File: kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/project/KieModuleServiceImpl.java [hunk 2]
```diff
@@ -117,6 +118,11 @@ private void checkRepositories(final POM pom) {
         }
     }
 
+    @Override
+    public KieModulePackages resolveModulePackages(Module activeModule) {
```
Comment (caponetto) [inline kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/project/KieModuleServiceImpl.java]: Consider creating at least one unit test for this public method.

File: kie-wb-common-services/kie-wb-common-services-api/src/main/java/org/kie/workbench/common/services/shared/project/KieModulePackages.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+@Portable
+public class KieModulePackages {
+
+    private Set<Package> packages;
+
+    private Package defaultPackage;
+
+    public KieModulePackages() { }
+
+    public KieModulePackages(Set<Package> packages, Package defaultPackage) {
```
[the remark this answers is not in the document]
Comment (pefernan) [reply kie-wb-common-services/kie-wb-common-services-api/src/main/java/org/kie/workbench/common/services/shared/project/KieModulePackages.java]: +1

**Comment by barboras7** [comment]
jenkins execute full downstream build

**Comment by barboras7** [comment]
jenkins execute full downstream build

**Comment by pefernan** [comment]
jenkins execute full downstream build

File: kie-wb-common-services/kie-wb-common-services-api/src/main/java/org/kie/workbench/common/services/shared/project/KieModulePackages.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import org.jboss.errai.common.client.api.annotations.Portable;
+
+@Portable
+public class KieModulePackages {
+
+    private Set<Package> packages;
+
+    private Package defaultPackage;
+
+    public KieModulePackages() { }
```
[the remark this answers is not in the document]
Comment (akumar074) [reply kie-wb-common-services/kie-wb-common-services-api/src/main/java/org/kie/workbench/common/services/shared/project/KieModulePackages.java]: Done.

File: kie-wb-common-services/kie-wb-common-services-api/src/main/java/org/kie/workbench/common/services/shared/project/KieModulePackages.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (akumar074) [reply kie-wb-common-services/kie-wb-common-services-api/src/main/java/org/kie/workbench/common/services/shared/project/KieModulePackages.java]: Done.

File: kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/project/KieModuleServiceImpl.java (hunk 2 above)
Comment (akumar074) [reply kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/project/KieModuleServiceImpl.java]: Thanks for the review comments, Added a test in ModuleServiceImplResolvePackagesTest.

**Comment by paulovmr** [comment]
jenkins execute full downstream build

**Comment by akumar074** [comment]
jenkins execute full downstream build

**Comment by barboras7** [comment]
jenkins execute full downstream build

**Comment by akumar074** [comment]
> Hi @akumar074,
> 
> could you please check why the full downstream build is failing? I would like to check this PR also manually and I am not able to download the .war file from full downstream.

Hi @barboras7 

There is some issue in the pipeline, it has been failing for all the PRs. I have uploaded the war [here](https://drive.google.com/open?id=1oFj4dIl_TzMGfHuJpRXMTgW6Va3TEvNX)

Thanks

**Comment by barboras7** [comment]
Thanks @akumar074, works as expected :+1:

**Comment by akumar074** [comment]
jenkins execute full downstream build

**Comment by akumar074** [comment]
jenkins retest this please

**Comment by pefernan** [comment]
@barboras7 due to the latest changes, would you mind review again?

**Comment by barboras7** [comment]
jenkins execute full downstream build

**Comment by akumar074** [comment]
@barboras7 @pefernan Please check deployable war [link](https://drive.google.com/open?id=11j9krqVd5xgBhoGw_sMBxAkGICkbzYTj) for the latest change.

**Comment by pefernan** [comment]
jenkins retest this please

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/knowledgebases/item/KnowledgeBaseItemView.java [hunk 5]
```diff
@@ -61,11 +65,11 @@
 
     @Inject
     @DataField("equals-behavior-select-container")
-    private HTMLDivElement equalsBehaviorSelectContainer;
+    private KieEnumSelectElement<AssertBehaviorOption> equalsBehaviorSelectContainer;
```
Comment (pefernan) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/knowledgebases/item/KnowledgeBaseItemView.java]: I'll be picky here... but the property name is not clear... `equalsBehaviorSelect` would be more appropiate than `equalsBehaviorSelectContainer`. The same would apply for  the @DataField... and the html template.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/knowledgebases/item/KnowledgeBaseItemView.java [hunk 6]
```diff
@@ -61,11 +65,11 @@
...
 
     @Inject
     @DataField("event-processing-model-select-container")
-    private HTMLDivElement eventProcessingModelSelectContainer;
+    private KieEnumSelectElement<EventProcessingOption> eventProcessingModelSelectContainer;
```
Comment (pefernan) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/knowledgebases/item/KnowledgeBaseItemView.java]: Same here.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/knowledgebases/item/knowledgesessions/KnowledgeSessionListItemView.java [hunk 7]
```diff
@@ -48,7 +51,7 @@
 
     @Inject
     @DataField("clock-select-container")
-    private HTMLDivElement clockSelectContainer;
+    private KieEnumSelectElement<ClockTypeOption> clockSelectContainer;
```
Comment (pefernan) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/knowledgebases/item/knowledgesessions/KnowledgeSessionListItemView.java]: Same here.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/knowledgebases/item/knowledgesessions/listener/ListenerListItemView.java [hunk 8]
```diff
@@ -21,19 +21,21 @@
...
+import org.kie.workbench.common.screens.library.client.settings.util.select.KieEnumSelectElement;
+import org.kie.workbench.common.services.shared.kmodule.ListenerModel;
 
 @Templated("#root")
 public class ListenerListItemView implements ListenerListItemPresenter.View {
 
     @Inject
     @DataField("kind-select-container")
-    private HTMLDivElement kindSelectContainer;
+    private KieEnumSelectElement<ListenerModel.Kind> kindSelectContainer;
```
Comment (pefernan) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/knowledgebases/item/knowledgesessions/listener/ListenerListItemView.java]: Same here.

File: kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/editor/DeploymentPopupViewImpl.java [hunk 9]
```diff
@@ -89,7 +89,7 @@
 
     @Inject
     @DataField
-    private HTMLDivElement serverTemplateDropdownContainer;
+    private KieSelectElement serverTemplateDropdownContainer;
```
Comment (pefernan) [inline kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/editor/DeploymentPopupViewImpl.java]: Same here.

File: kie-wb-common-widgets/kie-wb-common-ui/src/main/java/org/kie/workbench/common/widgets/client/widget/KieSelectElementView.html [hunk 10]
```diff
@@ -14,7 +14,7 @@
   ~ limitations under the License.
   -->
 
-<div class="btn-group bootstrap-select form-control" data-field="root">
+<div class="btn-group bootstrap-select form-control">
```
Comment (pefernan) [inline kie-wb-common-widgets/kie-wb-common-ui/src/main/java/org/kie/workbench/common/widgets/client/widget/KieSelectElementView.html]: Good trick here ;-)

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/knowledgebases/item/KnowledgeBaseItemView.java (hunk 5 above)
Comment (akumar074) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/knowledgebases/item/KnowledgeBaseItemView.java]: Thanks for the review. I have made changes accordingly.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/knowledgebases/item/KnowledgeBaseItemView.java (hunk 6 above)
Comment (akumar074) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/knowledgebases/item/KnowledgeBaseItemView.java]: Done

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/knowledgebases/item/knowledgesessions/KnowledgeSessionListItemView.java (hunk 7 above)
Comment (akumar074) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/knowledgebases/item/knowledgesessions/KnowledgeSessionListItemView.java]: Done

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/knowledgebases/item/knowledgesessions/listener/ListenerListItemView.java (hunk 8 above)
Comment (akumar074) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/knowledgebases/item/knowledgesessions/listener/ListenerListItemView.java]: Done

File: kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/editor/DeploymentPopupViewImpl.java (hunk 9 above)
Comment (akumar074) [reply kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/editor/DeploymentPopupViewImpl.java]: Done

**Comment by barboras7** [comment]
Sure @pefernan, done :+1:

**Comment by pefernan** [comment]
jenkins execute full downstream build please

**Comment by pefernan** [comment]
build failures appear to be unrelated

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by pefernan** [comment]
jenkins execute full downstream build please

**Comment by pefernan** [comment]
Downstream seems to fail due to unrelated test failures

**Comment by akumar074** [comment]
jenkins execute full downstream build please

**Comment by paulovmr** [comment]
jenkins execute full downstream build please

**Comment by pefernan** [comment]
@akumar074 great work buddy!

**Comment by akumar074** [comment]
> @akumar074 great work buddy!

Thank you @pefernan  :)

**Pull request merged**